src/object.jl

```diff
@@ -167,14 +167,20 @@ mutable struct PrefechedDownloadStream{T <: Object} <: IO
         )
         prefetch_size > 0 || throw(ArgumentError("`prefetch_size` must be positive, got $prefetch_size"))
         prefetch_multipart_size > 0 || throw(ArgumentError("`prefetch_multipart_size` must be positive, got $prefetch_multipart_size"))
```

Comment:
I'm not sure where the right place to comment on this code is. I came here from https://github.com/RelationalAI/raicode/pull/12053.

--------

Why aren't these assertions done before constructing the `io` object above? can we move them?